scripts/elf_helper.py

```diff
@@ -0,0 +1,522 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2017-2018 Linaro
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import sys
+import argparse
+import pprint
+import os
+import struct
+from distutils.version import LooseVersion
+
+import elftools
+from elftools.elf.elffile import ELFFile
+from elftools.dwarf import descriptions
+from elftools.elf.sections import SymbolTableSection
+
+if LooseVersion(elftools.__version__) < LooseVersion('0.24'):
+    sys.stderr.write("pyelftools is out of date, need version 0.24 or later\n")
+    sys.exit(1)
+
+
+def subsystem_to_enum(subsys):
+    return "K_OBJ_DRIVER_" + subsys[:-11].upper()
+
+
+def kobject_to_enum(ko):
+    return "K_OBJ_" + ko[2:].upper()
+
+
+DW_OP_addr = 0x3
+DW_OP_fbreg = 0x91
+STACK_TYPE = "_k_thread_stack_element"
+thread_counter = 0
+
+# Global type environment. Populated by pass 1.
+type_env = {}
+kobjects = {}
+subsystems = {}
+
+# --- debug stuff ---
+
+scr = os.path.basename(sys.argv[0])
+
+# --- type classes ----
+
+
+class KobjectInstance:
```

Comment:
gen_kobject_list.py declares these classes as well; can we move those to a separate file and only import them in this script? (Same for KobjectType, ArrayType, etc.)

Reply:
agreed

Reply:
Oh, disregard this comment. This is precisely what this file is.

Reply:
yeah i split the change into 2 separate patches.  1 to implement the helper, and the other to change usage of existing files over to use the new helper.